src/webui/abstractwebapplication.cpp

```diff
@@ -370,12 +370,12 @@ bool AbstractWebApplication::sessionEnd()
 
 QString AbstractWebApplication::saveTmpFile(const QByteArray &data)
 {
-    QTemporaryFile tmpfile(QDir::temp().absoluteFilePath("qBT-XXXXXX.torrent"));
-    tmpfile.setAutoRemove(false);
-    if (tmpfile.open()) {
-        tmpfile.write(data);
-        tmpfile.close();
-        return tmpfile.fileName();
+    QTemporaryFile *tmpfile = new QTemporaryFile(QDir::temp().absoluteFilePath("qBT-XXXXXX.torrent"));
+    tmpfile->setAutoRemove(false);
+    if (tmpfile->open()) {
+        tmpfile->write(data);
+        tmpfile->close();
+        return tmpfile->fileName();
```

Comment:
Cppcheck throws an error. It's a false positive but I think this is more readable.
> Brief: Return value of allocation function tmpfile is not stored.

Check **src\core\downloadthread.cpp** line **143**